assets/app/scripts/controllers/overview.js

```diff
@@ -238,12 +241,7 @@ angular.module('openshiftConsole')
       if (configJson) {
         try {
           var depConfig = $.parseJSON(configJson);
-          if (depConfig.apiVersion === "v1beta1") {
-            // TODO: will be slightly inaccurate for now
-            deployment.status = {details: depConfig.details};
-          } else {
-            deployment.status = depConfig.status;
-          }
+          deployment.details = depConfig.details || depConfig.status.details;
```

Comment:
scope this down to just the causes, since that's all the view cares about right now. move the extraction of causes into a filter, with a error logged if the deployment config is an api version we don't recognize.